sls-versions/src/main/java/com/palantir/sls/versions/VersionComparator.java

```diff
@@ -105,6 +110,17 @@ private int compareFirstSequenceVersions(OrderableSlsVersion left, OrderableSlsV
                 "Expected to find a first sequence number for version",
                 SafeArg.of("version", right.getValue()));
 
-        return Integer.compare(leftInt.getAsInt(), rightInt.getAsInt());
+        int firstSequenceCompare = Integer.compare(leftInt.getAsInt(), rightInt.getAsInt());
+        if (firstSequenceCompare != 0) {
+            return firstSequenceCompare;
+        }
+
+        return stringCompare(left, right);
+    }
+
+    private int stringCompare(OrderableSlsVersion left, OrderableSlsVersion right) {
+        // We know everything but the hash is equal between the two versions.
+        // We fall back to string based comparison to settle potential differences between git hashes.
+        return left.toString().compareTo(right.toString());
```

Comment:
one weird edge case that could bring us to this point is the case where someone has got leading zeros on their version string (e.g. "1.001.0-1-gbbb" vs "1.1.0-1-gaaa"),


but maybe that's fine to give this case an explicit order?

Reply:
It's an interesting case. If we already accept to give an order to git hashes, I would not be too concerned just picking some arbitrary (but stable) ordering convention for this.